feat: introduce sign mode for driver packaging, allowing skipping of signing steps (#649)
microsoft/windows-drivers-rs: crates/cargo-wdk/README.md

Options:
      --profile <PROFILE>          Build artifacts with the specified profile
      --target-arch <TARGET_ARCH>  Build for the target architecture
      --sign-mode <SIGN_MODE>      Driver signing mode [default: test] [possible values: off, test]
      --verify-signature           Verify the signature
      --sample                     Build sample class driver project

>   -h, --help                       Print help (see more with '--help')

Reviewer: Can you remove the `(see more with '--help')` part? It is a bit confusing in the readme.

Author: Fixed ✅

#### Signing and Verification

> The `--sign-mode` flag controls how driver artifacts are signed. It accepts the following values:

Reviewer: ```suggestion
The `build` command has a `--sign-mode` flag that controls how driver artifacts are signed. It accepts the following values:
```

Author: Fixed ✅

> - `test` (default): Sign with an auto-generated self-signed certificate. `build` looks for a certificate called `WDRLocalTestCert` in a store called `WDRTestCertStore`. Make sure you place your signing certificate there with that name. If no certificate is found, `build` will automatically generate a new self-signed one and add it for you.

Reviewer: ```suggestion
- `test` (default): Sign with a test certificate. The command looks for a certificate called `WDRLocalTestCert` in a store called `WDRTestCertStore`. If you wish you to use your certificate add it to the same store with the same name. Otherwise a self-signed certifcate will be automatically generated, added and used for signing.  
```

We can't say "sign with an auto-generated self-signed certificate" while also saying "make sure you place your own cert in the store". The more accurate description is "sign with a test certificate". That's all. It may or may not be auto generated.

I have also suggested changes in the rest of the paragraph for clarity.

Author: Fixed ✅

> - `off`: Skip signing entirely. No signing or signature verification steps are performed. This is useful when you intend to sign the driver later with your own toolchain (for example, HLK certification). Note that an unsigned driver cannot be installed on Windows without first enabling [test signing](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/the-testsigning-boot-configuration-option) or otherwise loading it through a signed mechanism.

Reviewer: ```suggestion
- `off`: Skip signing entirely. No signing or signature verification is performed. This is useful when you intend to sign the driver later with your own toolchain (e.g. HLK certification).
```

Developers know this already and need not be repeated here:

_Note that an unsigned driver cannot be installed on Windows without first enabling [test signing](https://learn.microsoft.com/en-us/windows-hardware/drivers/install/the-testsigning-boot-configuration-option) or otherwise loading it through a signed mechanism._

Replace "steps are" with "is" and "for example" with "e.g." for brevity.

Author: Fixed ✅

If the `--verify-signature` flag is provided, the signatures are verified after signing. For verification to work, make sure you add a copy of the signing certificate in the `Trusted Root Certification Authorities` store. For security reasons `build` does not automatically do this even when it automatically generates the cert. You will have to always perform this step manually.

> `--verify-signature` cannot be combined with `--sign-mode=off` because there is no signature to verify. Passing both will cause `build` to fail with an error.

Reviewer: ```suggestion
`--verify-signature` cannot be combined with `--sign-mode=off` because if signing is off there is nothing to verify. Passing both will cause `build` to fail with an error.
```

Makes the sentence clearer

Author: Fixed ✅

    ```pwsh
    cargo wdk build --target-arch amd64
    ```

> - To build a driver project and skip signing (for example, to sign later with a production certificate), run:

Reviewer: ```suggestion
- To build a driver project with test signing, navigate to the root of the project and run:
```
Changed "and skip signing" to "with test signing" as we should use the testing signing use-case as the example (see below suggestion the command line as well).

Secondly, remove the stuff in parenthesis as it is common knowledge and doesn't add much. Change "run" to "navigate to the root of the project and run" for consistency with other examples here.

Author: Fixed ✅

    ```pwsh

>     cargo wdk build --sign-mode test

Reviewer 2: nit: Since the default is test signing on. Its better to put the counter example here

```
- To build a driver project with signing off, navigate to the root of the project and run:

    ```pwsh
    cargo wdk build --sign-mode off
    ```
```

Reviewer: That's a good point. I think I had asked @svasista-ms to change it from `off` to `test` earlier. @svasista-ms apologies, but I think it's better to go with the `off` example.

>     cargo wdk build --sign-mode off

Reviewer: ```suggestion
    cargo wdk build --sign-mode test
```

Test signing is much more common than disabling off signing so we should use that as the example

Author: Fixed ✅
